docs/GettingStarted.md

```diff
@@ -269,4 +276,81 @@ Patches are submitted to LLVM/MLIR via GitHub pull-requests, the basic flow is a
 6) Publish the branch on your fork of the repository and create a GitHub pull-request.
 
 When your review converges and your patch is approved, it can be merged directly on GitHub.
-If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.
+If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.  
+  
+## Writing a basic CIRCT Pass  
+Passes can be added at several levels in CIRCT. Here we illustrate this with a simple pass, added 
+to the `firrtl` dialect, that replaces all of the wire names with `foo`: 
+1) Update your dialect’s Passes.td file to define what your pass will do in a high-level, well documented way, 
+e.g. in `include/circt/Dialect/FIRRTL/Passes.td`:  
+```
+def FooWires : Pass<"firrtl-foo-wires", "firrtl::CircuitOp"> {
+  let summary = "Change all wires' name to foo_<n>.";
+  let description = [{
+    Very basic pass that numbers all of the wires in a given module.
+    The wires' names are then all converte to foo_<that number>.
+  }];
+  let constructor = "circt::firrtl::createFooWiresPass()";
+  let statistics = [
+    Statistic<"numChangedWires", "num-changed-wires", "Number of wires changed">
+  ];
+}
+```    
+Once this is added, compile CIRCT. This will generate a base class for your pass following the naming
+defined in the tablegen description. This base class will contain some methods that need to be implemented.
+Your goal is to now implement those.  
+
+2) Create a new file that contains your pass. Don't forget to also add it in the folder's CMakeLists.txt. 
+This file should implement:  
+  - A struct with the name of your pass, which extends the pass base class generated by the tablegen description.
+  - This struct should define an override of the `void runOnOperation()` method, e.g.  
+  ```cpp
+  namespace {
+      // A test pass that simply replaces all wire names with foo_<n>
+      struct FooWires : FooWiresBase<FooWires> {
+        void runOnOperation() override;
+      };
+  }
+  ```  
+  - The `runOnOperation` method will contain the actual logic of your pass, which you can now implement, e.g.
+  ```cpp
+  void FooWires::runOnOperation() {
+    size_t nWires = 0; // Counts the number of wires modified
+    module.walk([&](hw::WireOp wire) { // Walk over every wire in the module
+      wire.setName("foo_" + std::to_string(nWires++)); // Rename said wire
+    });
+  }
+  ``` 
+  Here `module.walk([&](WireOp wire) { ... });` is used to traverse every wire in the design, you can also
+  use it to generically walk over all operations and then visit them individually using a [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) by overloading type visitors defined by each dialect, e.g.
```

Comment:
Not sure the audience of the tutorial, but they might find this link useful: https://mlir.llvm.org/docs/Tutorials/UnderstandingTheIRStructure/

Reply:
Compiler engineers that don't know MLIR but want to get a quick and dirty start in CIRCT so that they can easily test their ideas out (i.e. me about a year ago lol). I'll add that link plus a few others scattered around so that people also have access to more detailed tutorials if they want them.